src/external_cli/rustup.rs

```diff
@@ -0,0 +1,49 @@
+//! Utilities for the `rustup` CLI tool.
+
+#![expect(dead_code, reason = "Will be used for build/run commands")]
+
+use std::process::{exit, Command};
+
+use dialoguer::Confirm;
+
+const PROGRAM: &str = "rustup";
+
+/// Given a target triple, determine if it is already installed.
+fn is_target_installed(target: &str) -> bool {
+    let output = Command::new(PROGRAM).arg("target").arg("list").output();
+
+    // Check if the target list has an entry like this:
+    // <target_triple> (installed)
+    let Ok(output) = output else { return false };
+    let Ok(list) = String::from_utf8(output.stdout) else {
+        return false;
+    };
```

Comment:
This could be replaced with [`String::from_utf8_lossy()`](https://doc.rust-lang.org/stable/std/string/struct.String.html#method.from_utf8_lossy), which replaces unknown unicode with [`U+FFFD`](https://doc.rust-lang.org/stable/std/char/constant.REPLACEMENT_CHARACTER.html) instead of erroring. This would make it more resistant to breakage, in case someone installs a custom target with an invalid Unicode name.